src/axom/spin/tests/spin_bvh.cpp

```diff
@@ -0,0 +1,540 @@
+  }
+
+  // traverse the BVH to find the candidates for all the centroids
+  IndexType* offsets    = axom::allocate< IndexType >( ncells );
+  IndexType* counts     = axom::allocate< IndexType >( ncells );
+  IndexType* candidates = nullptr;
+  bvh.find( offsets, counts, candidates, ncells, xc, yc, zc );
+
+  EXPECT_TRUE( candidates != nullptr );
+
+  spin::UniformGrid< IndexType, NDIMS > ug( lo, hi, res );
+
+  for ( IndexType i=0; i < ncells; ++i )
+  {
+    PointType q            = PointType::make_point( xc[ i ],yc[ i ],zc[ i ] );
+    const int donorCellIdx = ug.getBinIndex( q );
+    EXPECT_EQ( counts[ i ], 1 );
+    EXPECT_EQ( donorCellIdx, candidates[ offsets[ i ] ] );
+  } // END for all cell centroids
+
+  axom::deallocate( candidates );
+
+  // check points that are outside by shifting the query points
+  constexpr double OFFSET = 10.0;
+  for ( IndexType i=0; i < ncells; ++i )
+  {
+     xc[ i ] += OFFSET;
+     yc[ i ] += OFFSET;
+     zc[ i ] += OFFSET;
+  }
+
+  bvh.find( offsets, counts, candidates, ncells, xc, yc, zc );
+
+  for ( IndexType i=0; i < ncells; ++i )
+  {
+    EXPECT_EQ( counts[ i ], 0 );
+  }
+  EXPECT_TRUE( candidates == nullptr );
+
+  axom::deallocate( offsets );
+  axom::deallocate( candidates );
+  axom::deallocate( counts );
+  axom::deallocate( aabbs );
+
+  axom::setDefaultAllocator( current_allocator );
+}
+
+//------------------------------------------------------------------------------
+template < typename ExecSpace, typename FloatType >
+void check_find2d( )
+{
+  constexpr int NDIMS   = 2;
+  constexpr IndexType N = 4;
+
+  umpire::Allocator current_allocator = axom::getDefaultAllocator();
+  axom::setDefaultAllocator( spin::execution_space<ExecSpace>::allocatorID());
+
+  using PointType = primal::Point< double, NDIMS >;
+
+  double lo[ NDIMS ] = { 0.0, 0.0 };
+  double hi[ NDIMS ] = { 3.0, 3.0 };
+  int res[ NDIMS ]   = { N-1, N-1 };
+
+  mint::UniformMesh mesh( lo, hi, N, N );
+  FloatType* xc = mesh.createField< FloatType >( "xc", mint::CELL_CENTERED );
+  FloatType* yc = mesh.createField< FloatType >( "yc", mint::CELL_CENTERED );
+  const IndexType ncells = mesh.getNumberOfCells();
+
+  FloatType* aabbs = nullptr;
+  generate_aabbs_and_centroids2d( &mesh, aabbs, xc, yc );
+
+  // construct the BVH
+  spin::BVH< NDIMS, ExecSpace, FloatType > bvh( aabbs, ncells );
+  bvh.build( );
+
+  FloatType min[ NDIMS ];
+  FloatType max[ NDIMS ];
+  bvh.getBounds( min, max );
+  for ( int i=0; i < NDIMS; ++i )
+  {
+    EXPECT_DOUBLE_EQ( min[ i ], lo[ i ] );
+    EXPECT_DOUBLE_EQ( max[ i ], hi[ i ] );
+  }
+
+  // traverse the BVH to find the candidates for all the centroids
+  IndexType* offsets    = axom::allocate< IndexType >( ncells );
+  IndexType* counts     = axom::allocate< IndexType >( ncells );
+  IndexType* candidates = nullptr;
+  bvh.find( offsets, counts, candidates, ncells, xc, yc );
+
+  EXPECT_TRUE( candidates != nullptr );
+
+  spin::UniformGrid< IndexType, NDIMS > ug( lo, hi, res );
```

Comment:
so the trick for these cases is that you know the input is a uniform grid, so you query using BVH and then create a spin uniform grid data to check against?

Would it be better to test the BVH results more stand alone w/o using the sping UniformGrid -- b/c then bugs in the UniformGrid won't undermine our BVH testing. 

Or maybe comments that describe the test data would be helpful so it's easier to for folks to understand the nature of the test down the road

Reply:
Yes, that's exactly right. I am checking the output against the `spin::UniformGrid`. Since the input points are the cell-centers of a uniform mesh, the `BVH` returns the same results as the `spin::UniformGrid` in this case.

I see the point of a bug in the UniformGrid percolating into the BVH tests  and producing false positives or negatives. I prefer stand-alone tests, but, in this case, I would have to roll similar functionality as to what the UniformGrid already provides for this test.

I'll definitely add comments for now and think some more about easy ways to make the tests herein more stand-alone.

Reply:
thanks for confirming, adding comments sounds like a good way to address

Reply:
Done.